rosco/rosco-core/src/test/java/com/netflix/spinnaker/rosco/services/ClouddriverServiceTest.java

```diff
@@ -0,0 +1,130 @@
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.catchThrowable;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import com.netflix.spectator.api.NoopRegistry;
+import com.netflix.spinnaker.config.OkHttp3ClientConfiguration;
+import com.netflix.spinnaker.config.OkHttpMetricsInterceptorProperties;
+import com.netflix.spinnaker.config.RetrofitConfiguration;
+import com.netflix.spinnaker.config.okhttp3.RawOkHttpClientFactory;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import com.netflix.spinnaker.kork.retrofit.Retrofit2SyncCall;
+import com.netflix.spinnaker.kork.retrofit.exceptions.SpinnakerHttpException;
+import com.netflix.spinnaker.okhttp.OkHttp3MetricsInterceptor;
+import com.netflix.spinnaker.okhttp.OkHttpClientConfigurationProperties;
+import com.netflix.spinnaker.okhttp.Retrofit2EncodeCorrectionInterceptor;
+import com.netflix.spinnaker.okhttp.SpinnakerRequestHeaderInterceptor;
+import com.netflix.spinnaker.okhttp.SpinnakerRequestInterceptor;
+import com.netflix.spinnaker.retrofit.Retrofit2ConfigurationProperties;
+import com.netflix.spinnaker.retrofit.RetrofitConfigurationProperties;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import retrofit.RequestInterceptor;
+
+@SpringBootTest(
+    classes = {
+      ServiceConfig.class,
+      ClouddriverServiceTest.TestConfig.class,
+      DynamicConfigService.NoopDynamicConfig.class,
+      NoopRegistry.class,
+      Retrofit2EncodeCorrectionInterceptor.class,
+      RetrofitConfiguration.class,
+      Retrofit2ConfigurationProperties.class,
+      OkHttp3MetricsInterceptor.class,
+      RawOkHttpClientFactory.class,
+      RetrofitConfigurationProperties.class,
+      OkHttpClientConfigurationProperties.class,
+      OkHttpMetricsInterceptorProperties.class,
+      ObjectMapper.class,
+      OkHttp3ClientConfiguration.class
+    })
+public class ClouddriverServiceTest {
+  @RegisterExtension
+  static WireMockExtension wmClouddriver =
+      WireMockExtension.newInstance().options(wireMockConfig().dynamicPort()).build();
+
+  @Autowired ClouddriverService clouddriverService;
+
+  static String WM_CORRECT_ENDPOINT =
+      "/api/artifacts/fetch/"; // since the baseUrl has /api/ at the end.
```

Comment:
I don't understand this variable name/comment.  Isn't the correct endpoint `/artifacts/fetch` from clouddriver's [ArtifactController](https://github.com/spinnaker/spinnaker/blob/a5c65ed59d9eab54f9abdb01ba6f2bbdc46d0df8/clouddriver/clouddriver-web/src/main/java/com/netflix/spinnaker/clouddriver/controllers/ArtifactController.java#L76)?

Reply:
For wiremock, the end point starts from the first slash in the final URL so with the base url: http://localhost:<port>/api/  and endpoint: /artifacts/fetch the wiremock's endpoint becomes /api/artifacts/fetch.  But since retrofit2 removes /api from the final URL(http://localhost:<port>/artifacts/fetch) when leading slash exists on the endpoint, the wiremock's endpoint becomes /artifacts/fetch and this happened to be same as couddriver's endpoint on the interface method but in the given context(base url having multiple slashes) it's incorrect.

Reply:
OK, we talked and I think I understand this better.  `/api` is relevant because this test explicitly adds it.  And in the real world it would take some spring boot config property on the clouddriver side to match, or perhaps running some reverse proxy, etc. in front of clouddriver.  And the point of the test is to exercise this case, where the base URL has a path component after the hostname.